### PR TITLE
Doc fixes: remove typos and long sentence

### PR DESCRIPTION
**Pull request by ms42Q**
**What this PR does / why we need it**:
Minor fix in the docs

### CODE REVIEW

File: docs/sources/clients/promtail/configuration.md [hunk 1]
```diff
@@ -336,10 +336,9 @@ kubernetes_sd_configs:
 
 ### pipeline_stages
 
-The [pipeline](../pipelines/) stages (`pipeline_stages`) is used to transform
-log entries and their labels after discovery and consists of a list of any of the items listed below.
+[Pipeline](../pipelines/) stages are used to transform log entries and their labels. The pipline is executed after the discovery process finishes. The `pipeline_stages` object consists of a list of stages which correspond to the items listed below.
```
Comment (oddlittlebird) [inline docs/sources/clients/promtail/configuration.md:339]:
```suggestion
[Pipeline](../pipelines/) stages are used to transform log entries and their labels. The pipeline is executed after the discovery process finishes. The `pipeline_stages` object consists of a list of stages which correspond to the items listed below.
```

File: docs/sources/clients/promtail/configuration.md [hunk 2]
```diff
@@ -336,10 +336,9 @@ kubernetes_sd_configs:
...
 
-Stages serve several purposes, more detail can be found [here](../pipelines/), however generally you extract data with `regex` or `json` stages into a temporary map which can then be use as `labels` or `output` or any of the other stages aside from `docker` and `cri` which are explained in more detail below.
+Stages serve several purposes, more detail can be found [here](../pipelines/). However in most cases you extract data from logs with `regex` or `json` stages. The extracted data is transformed into a temporary map object. The data can then be used by promtail e.g. as values for `labels` or as an `output`. Additionally any other stage aside from `docker` and `cri` can access the extracted data.
```
Comment (oddlittlebird) [inline docs/sources/clients/promtail/configuration.md:341]:
```suggestion
Stages serve several purposes, more detail can be found [here](../pipelines/).

In most cases, you extract data from logs with `regex` or `json` stages. The extracted data is transformed into a temporary map object. The data can then be used by promtail, e.g. as values for `labels` or as an `output`. Additionally any other stage aside from `docker` and `cri` can access the extracted data.
```

Comment (oddlittlebird) [reply docs/sources/clients/promtail/configuration.md:341]: Best practice is to keep paragraphs to three sentences or left when writing for online viewing.

**Comment by ms42Q** [comment]
Thanks for your review @oddlittlebird !

I applied your suggestions and rebased

**Comment by oddlittlebird** [comment]
Thanks for contributing to the Loki docs!

**Pull request merged**